web/src/components/storage/PartitionPage.tsx

```diff
@@ -0,0 +1,686 @@
+  target: string;
+};
+
+function SizeOptions({ mountPoint, target }: SizeOptionsProps): React.ReactNode {
+  return (
+    <SelectList>
+      {mountPoint === NO_VALUE && (
+        <SelectOption value={NO_VALUE}>
+          <SizeOptionLabel value={NO_VALUE} mountPoint={mountPoint} target={target} />
+        </SelectOption>
+      )}
+      {mountPoint !== NO_VALUE && target !== NEW_PARTITION && (
+        <SelectOption value={NO_VALUE} description={sprintf(_("Keep %s size"), target)}>
+          <SizeOptionLabel value={NO_VALUE} mountPoint={mountPoint} target={target} />
+        </SelectOption>
+      )}
+      {mountPoint !== NO_VALUE && target === NEW_PARTITION && (
+        <>
+          <SelectOption value="auto" description={_("A proper size is automatically calculated")}>
+            <SizeOptionLabel value="auto" mountPoint={mountPoint} target={target} />
+          </SelectOption>
+          <SelectOption
+            value="custom"
+            description={_("Define a custom size between a minimum and a maximum")}
+          >
+            <SizeOptionLabel value="custom" mountPoint={mountPoint} target={target} />
+          </SelectOption>
+        </>
+      )}
+    </SelectList>
+  );
+}
+
+type AutoSizeProps = {
+  mountPoint: string;
+};
+
+function AutoSize({ mountPoint }: AutoSizeProps): React.ReactNode {
+  const volume = useVolume(mountPoint);
+
+  const conditions = [];
+
+  if (volume.outline.snapshotsAffectSizes)
+    // TRANSLATORS: item which affects the final computed partition size
+    conditions.push(_("the configuration of snapshots"));
+
+  if (volume.outline.sizeRelevantVolumes && volume.outline.sizeRelevantVolumes.length > 0)
+    // TRANSLATORS: item which affects the final computed partition size
+    // %s is replaced by a list of mount points like "/home, /boot"
+    conditions.push(
+      sprintf(
+        _("the presence of the file system for %s"),
+        // TRANSLATORS: conjunction for merging two list items
+        volume.outline.sizeRelevantVolumes.join(_(", ")),
+      ),
+    );
+
+  if (volume.outline.adjustByRam)
+    // TRANSLATORS: item which affects the final computed partition size
+    conditions.push(_("the amount of RAM in the system"));
+
+  if (!conditions.length) return null;
+
+  // TRANSLATORS: the %s is replaced by the items which affect the computed size
+  const conditionsText = sprintf(
+    _("The final size for %s depends on %s."),
+    mountPoint,
+    // TRANSLATORS: conjunction for merging two texts
+    conditions.join(_(" and ")),
+  );
+
+  return <Alert variant="info" isInline isPlain title={conditionsText} />;
+}
+
+type CustomSizeOptionLabelProps = {
+  value: CustomSizeValue;
+};
+
+function CustonSizeOptionLabel({ value }: CustomSizeOptionLabelProps): React.ReactNode {
+  const labels = {
+    fixed: _("None"),
+    unlimited: _("Unlimited"),
+    range: _("Limited"),
+  };
+
+  return labels[value];
+}
+
+function CustonSizeOptions(): React.ReactNode {
```

Comment:
NP: typo, Custom instead of Custo**n**

Reply:
That comes from the target branch. Got visible here just due to an artifact caused by `push -f` on that branch.